src/main/scala/com/codacy/plugins/api/results/Pattern.scala

```diff
@@ -28,95 +28,28 @@ object Pattern {
 
   case class Definition(patternId: Pattern.Id, parameters: Option[Set[Parameter.Definition]])
 
-  trait Specification {
-    val patternId: Pattern.Id
-    val level: Result.Level
-    val category: Category
-    val subcategory: Option[Subcategory]
-    val parameters: Option[Set[Parameter.Specification]]
-    val languages: Option[Set[Language]]
+  case class Specification(patternId: Pattern.Id,
+                           level: Result.Level,
+                           category: Specification.Category,
+                           subcategory: Option[Specification.Subcategory],
+                           parameters: Option[Set[Parameter.Specification]],
+                           languages: Option[Set[Language]] = None) {
+    require(subcategory.isEmpty || category == Specification.Category.Security,
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Usage of untyped comparison operator](https://app.codacy.com/gh/codacy/codacy-plugins-api/pullRequest?prid=4817430)